File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp=[-math.inf for i in range(len(nums))] # dp represents the number of sequence upto this interval
        dp[0]=1
        for i in range(1,len(dp)):
            dp[i]=1
            for j in range(0,i):
                if nums[i]>nums[j]:
                    
                    if dp[j]+1>dp[i]:
                        dp[i]=1+dp[j]
        print(dp)
        return max(dp)

File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
     int[] dp=new int[nums.length];
    dp[0]=1;
    int max_count=1;
    for (int i=1;i<dp.length;i++){
        dp[i]=1;
        for (int j=0;j<i;j++){
            if (nums[i]>nums[j] && dp[j]+1>dp[i]){
                dp[i]=1+dp[j];
                max_count=Math.max(max_count,dp[i]);
            }
        }
    }
    return max_count;
    }
}